### PR TITLE
feat: Add option to not modify readonly files during copy, add logging of time

### LINKED ISSUE #19866: TaskCopyLocalFrontendFiles#copyLocalResources(File, File, String... ) takes a long time using Windows-OS
### Description of the bug

I'm using a windows pc and have noticed, that starting up our app takes longer than my colleges with mac.
The method com.vaadin.flow.server.frontend.TaskCopyLocalFrontendFiles#copyLocalResources takes up to a minute on windows and only 5 seconds on OSX.

We have found out, that the part that takes long ist what was added with https://github.com/vaadin/flow/issues/12711 . To be precise the code lines:
`try (Stream<Path> fileStream = Files
.walk(Paths.get(target.getPath()))) {
// used with try-with-resources as defined in walk API note
fileStream.filter(file -> !Files.isWritable(file)).forEach(
filePath -> filePath.toFile().setWritable(true));
}`

### Expected behavior

Could you please add an option/property to deactive the functionality of setting all files writeable.
None of our files are readonly and this is costing our developers lots of time.

### Minimal reproducible example

not sure

### Versions

- Vaadin / Flow version: 24.4.4
- Java version: 17.0.9_9
- OS version: Windows 10 Pro 10.0.19045
- IDE (if applicable): IDEA 2024.1.4

### PR DESCRIPTION
**Pull request by TatuLund**
Allows to skip setting the writable flag on copied files, by providing the vaadin.frontend.disableWritableFlagCheckOnCopy system property.
This may improve performance in certain scenarios with Windows OS.

Fixes #19866

### CODE REVIEW

**Comment by thomasharre** [comment]
@TatuLund you have changed the default behaviour, I only wanted a possibility to turn it of.
 I suggest changing:
`PREVENT_READONLY_FILES = true;`

**Comment by TatuLund** [comment]
@thomasharre Thanks for spotting that. I was testing this locally changing the value back and forth, it was wrong value by accident.

**Comment by mcollovati** [comment]
What's the idea behind the usage of the static field? Is the developer supposed to change it in the application code?
If so, this will work for development mode, but will not help when the task is executed by Maven or Gradle.
Could a system property be a better option?

**Comment by TatuLund** [comment]
> Could a system property be a better option?

Yes it would.

**Comment by thomasharre** [comment]
We are only experiencing problems in certain developer environements.
Therefor the static field is the most pragmatic solution for us.

A system property would also work, but the current solution with public static field would be a fast way to help us.

**Comment by mcollovati** [comment]
I still would prefer a system property, instead of the static field. Something like `vaadin.frontend.disableReadonlyCheckOnCopy` (feel free to suggest a better name), to make explicit that we are disabling an operation that was added to fix application failures (for example #12711)

**Comment by thomasharre** [comment]
A system property would also work fine for me.

So the code in copyLocalResources would be something like?
`boolean performReadonlyCheck =  !Boolean.valueOf(System.getProperty("vaadin.frontend.disableReadonlyCheckOnCopy", "false"));
if (performReadonlyCheck){
...
}
`

@TatuLund would that suit you? If yes, could you please commit that?

**Comment by thomasharre** [comment]
@TatuLund the member 'performReadonlyCheck' should not be static

**Comment by TatuLund** [comment]
> the member 'performReadonlyCheck' should not be static

@thomasharre No, it must be as the method where it is used is static

**Comment by thomasharre** [comment]
setting a static member in the constructor can't be right
then i would delete the member and introduce a new static method:

`  private static boolean performReadonlyCheck(){
    return !Boolean.valueOf(System.getProperty("vaadin.frontend.disableReadonlyCheckOnCopy", "false"));
  } `

**Comment by thomasharre** [comment]
@TatuLund thanks

@mcollovati could you take another look and descide if this can be merged?

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskCopyLocalFrontendFiles.java [hunk 1]
```diff
@@ -55,6 +55,11 @@ public class TaskCopyLocalFrontendFiles
         this.options = options;
     }
 
+    private static boolean isperformReadonly() {
```
Comment (mcollovati) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskCopyLocalFrontendFiles.java:58]: nit: I would rename this method to something more explicit, like `shouldApplyWriteableFlag()`.
Maybe also rename the system property to `vaadin.frontend.disableWritableFlagCheckOnCopy`

Not a blocker, anyway.

**Review by mcollovati — DISMISSED** [review]
Just a comment about naming, other than that LGTM

**Review by mcollovati — APPROVED** [review]

**Pull request merged**